ibm/resource_ibm_cis_filter.go

```diff
@@ -0,0 +1,305 @@
+)
+
+const (
+	ibmCISFilters        = "ibm_cis_filter"
+	cisFilterExpression  = "expression"
+	cisFilterPaused      = "paused"
+	cisFilterDescription = "description"
+	cisFilterID          = "filterid"
+)
+
+type FilterCreateUpdate struct {
+	Success bool `json:"success"`
+	Errors  []struct {
+	} `json:"errors"`
+	Messages []struct {
+	} `json:"messages"`
+	Result []struct {
+		ID          string `json:"id"`
+		Paused      bool   `json:"paused"`
+		Description string `json:"description"`
+		Expression  string `json:"expression"`
+		CreatedOn   struct {
+		} `json:"created_on"`
+		ModifiedOn struct {
+		} `json:"modified_on"`
+	} `json:"result"`
+}
+type FilterRead struct {
+	Success bool `json:"success"`
+	Errors  []struct {
+	} `json:"errors"`
+	Messages []struct {
+	} `json:"messages"`
+	Result struct {
+		ID          string `json:"id"`
+		Paused      bool   `json:"paused"`
+		Description string `json:"description"`
+		Expression  string `json:"expression"`
+		CreatedOn   struct {
+		} `json:"created_on"`
+		ModifiedOn struct {
+		} `json:"modified_on"`
+	} `json:"result"`
+}
+
+func resourceIBMCISFilter() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMCISFilterCreate,
+		Read:     resourceIBMCISFilterRead,
+		Update:   resourceIBMCISFilterUpdate,
+		Delete:   resourceIBMCISFilterDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisFilterPaused: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Filter Paused",
+			},
+			cisFilterID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Filter ID",
+			},
+			cisFilterExpression: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Filter Expression",
+			},
+			cisFilterDescription: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Filter Description",
+				ValidateFunc: InvokeValidator(ibmCISFilters, cisFilterDescription),
+			},
+		},
+	}
+}
+func resourceIBMCISFilterCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return fmt.Errorf("Error while Getting IAM Access Token using BluemixSession %s", err)
+	}
+	xAuthtoken := sess.Config.IAMAccessToken
+
+	cisClient, err := meta.(ClientSession).CisFiltersSession()
+	if err != nil {
+		return fmt.Errorf("Error while getting the CisFiltersSession %s", err)
+	}
+
+	crn := d.Get(cisID).(string)
+	zoneID, _, err := convertTftoCisTwoVar(d.Get(cisDomainID).(string))
+
+	var newfilter filtersv1.FilterInput
+
+	if _, ok := d.GetOk(cisFilterPaused); ok {
```

Comment:
```
if _, ok := d.GetOk(cisFilterPaused); ok {
    paused := d.Get(cisFilterPaused).(bool)
    newfilter.Paused = &paused
 }
```
Avoid Get multiple times.. This applies for all d.GetOK() scenerios in this code
```
if p, ok := d.GetOk(cisFilterPaused); ok {
    paused := p.(bool)
    newfilter.Paused = &paused
 }
```